core/packages/contracts/src/SovereignTreasury.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/AccessControl.sol";
+import "./IVault.sol";
+
+contract SovereignTreasury is AccessControl {
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant SENDER_ROLE = keccak256("SENDER_ROLE");
+
+    IVault public vault;
+
+    struct Message {
+        Action action;
+        bytes payload;
+    }
```

Comment:
type could go in some kinda PeeredApp base class, as mentioned in https://github.com/Snowfork/snowbridge/pull/781/files#r1155020081 ?